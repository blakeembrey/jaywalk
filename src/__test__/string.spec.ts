import { validationFunction } from '../'

describe('String', () => {
  it('should validate strings', () => {
    const validate = validationFunction({
      '@type': 'String'
    })

    const input = 'testing'

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should support formats', () => {
    const validate = validationFunction({
      '@type': 'String',
      format: 'uri'
    })

    const input = 'http://example.com'

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should reject non-strings', () => {
    const validate = validationFunction({
      '@type': 'String'
    })

    expect.assertions(1)

    return validate(10).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })
})
