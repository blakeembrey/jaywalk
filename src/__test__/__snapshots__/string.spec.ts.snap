// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String should reject non-strings 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "not a string",
    "schemaPath": Array [
      "@type",
    ],
  },
]
`;
